Updating reverse.c example for latest tool chain (clang-11) and dfx 0.6.10 (#49)
dfinity/examples: c/reverse/README.adoc

DFINITY uses standard wasm binaries, so we may use standard tools such as Clang
to build apps.

# Running in Docker

To create a docker container with all the libraries required and dfx follow the below instructions.  If you already have a debian environment you'd like to use, skip to the following section.

 Create a docker file with the below text:
 
  FROM debian
  RUN apt-get update
  RUN apt-get install -y curl gnupg xz-utils nodejs npm wget git

>   RUN apt install -y lsb-release wget software-properties-common

Reviewer: Are these required by llvm? dfx should be standalone and not depend on other packages. Pretty sure we don't need `nodejs` and `npm`, as there is no front end here.

Author: I had to install them to get some things to work.  Basically dfx build was failing when it was trying to build some things.  I can't find the error now, but it was right after the output of something like 'setting up the environment'.  My understanding was that node was required for dfx. Maybe that has changed in the latest release?  I'll try to replicate with out those packages and see.

Reviewer: With the new dfx.json file, `dfx build` is simply copying wasm files around. It won't do any actual build.

Author: Your right! You don't need node for this example so I removed it. If you want to play around with any other dfx commands like new you do need it or you get:

Node could not be found. Skipping installing the frontend example code.

I've removed them for the time being.

  RUN bash -c "$(wget -O - https://apt.llvm.org/llvm.sh)"
  CMD ["/bin/bash"]
  WORKDIR /usr/src
 
Create the new docker container by running the following command 

  $ docker build --tag debian:dfx .

Start the instance with 

  $ docker run -ti --name dfx debian:dfx
  
This will launch you into your instance. Now we need to pull the code by running:

  $ cd usr/src
  $ git clone https://github.com/dfinity/examples.git
  

> Install the dfinity sdk using

Reviewer 2: ```suggestion
Install the DFINITY sdk using
```

  $ sh -ci "$(curl -fsSL https://sdk.dfinity.org/install.sh)"
  
# Compiling the C code  

> If you are not using the docker container you need to nstall Clang:

Reviewer 2: ```suggestion
If you are not using the docker container you need to install Clang:
```

RUN bash -c "$(wget -O - https://apt.llvm.org/llvm.sh)"

> and set up a couple of aliases. navigate to the examples/c/reverse directory. Set up the following aliases:

Reviewer 2: ```suggestion
and set up a couple of aliases. Navigate to the examples/c/reverse directory. Set up the following aliases:
```

>   $ alias wld="wasm-ld-11 --no-entry --export-dynamic --allow-undefined"
>   $ alias wcc="clang-11 --target=wasm32 -c -O3"

Reviewer 2: ```suggestion
``` sh
  $ alias wld="wasm-ld-11 --no-entry --export-dynamic --allow-undefined"
  $ alias wcc="clang-11 --target=wasm32 -c -O3"
```
```

  $ wcc reverse.c
  $ wld reverse.o -o reverse.wasm

Now we need to build the dfx package.  The dfx.json and reverse.did that you need are provided in this repo.  Much of this is usually taken care of when you issue a dfx new command.

  $ dfx canister create reverse
  $ dfx build
  

> You now need a seperate terminal (docker exec -ti dfx bash) to start the dfx server. Make sure you are in the reverse directory and run

Reviewer 2: ```suggestion
You now need a separate terminal (`docker exec -ti dfx bash`) to start the local replica. Make sure you are in the reverse directory and run
```
